src/ast.rs

```diff
@@ -368,26 +377,40 @@ impl PartialOrd for RawBinaryOperator {
     }
 }
 
+/// This serves a similar purpose to ParenthesizedExpression.
+///
+/// see <https://google.github.io/xls/dslx_reference/#block-expressions>
+pub struct BlockExpression(pub Expression);
+
 /// *Part* of a let binding: the variable declaration and value it is bound to. What is missing
 /// is the expression in which the bound name is in-scope.
+///
+/// see <https://google.github.io/xls/dslx_reference/#let-expression>
 #[derive(Debug, PartialEq, Clone)]
 pub struct RawLetBinding {
     pub variable_declaration: VariableDeclaration,
     pub value: Box<Expression>,
 }
 pub type LetBinding = Spanned<RawLetBinding>;
 
-/// This struct exists to ensure that `From<Expression> for RawExpression` does not exist
-/// (because instead we have `From<ParenthesizedExpression> for RawExpression`). The former was
-/// bug prone: I was accidentally and unknowingly calling `from(Expression) -> RawExpression`.
-/// Inside the `from` we will discard the ParenthesizedExpression 'wrapper'.
+/// The condition and consequent of the first half of an `if` expression:
+/// `if condition { consequent }`
```

Comment:
Curious why needed. This isn't allowed:

```
let bleh = if foo { bleh };
```

So why is this needed vs. just:

```
pub struct RawConditional {
  pub condition: Expr,
  pub consequent: Expr,
  pub alternate: Expr,
}
```

Reply:
I think it's best explained if you look down to the relevant case of Expression. Also, I didn't want a recursive/nested if elseif structure. Just like for let bindings, I wanted a flat list. I assume in your suggestion above, alternate would be a Conditional expression?



    /// 1 or more `if` and `else if` expressions, followed by the final `else`'s alternate
    /// expression. I.e., if all the conditions in the vector are `false`, this `if...else
    /// if...else` expression evalutes to the alternate expression.
    IfElse(NonEmpty<Box<ConditionConsequent>>, Box<Expression>),